app/src/pages/DustyEthLockPage.tsx

```diff
@@ -0,0 +1,215 @@
+    networkType: string;
+    isProcessing: boolean;
+    allLockEvents: LockEvent[];
+    error: null;
+    fetchingLockData: boolean;
+}
+
+// need an empty interface to use states (React's generic positioning)
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface PageProps {}
+
+toast.configure({
+    position: 'top-right',
+    autoClose: 5000,
+    hideProgressBar: false,
+    closeOnClick: true,
+    pauseOnHover: true,
+    draggable: true,
+});
+
+class DustyEthLockPage extends React.Component<PageProps, PageStates> {
+    constructor(props: PageProps) {
+        super(props);
+        // initialize with null values
+        this.state = {
+            web3: {} as Web3,
+            accounts: [''],
+            contract: {} as Contract,
+            isLoading: true,
+            networkType: '',
+            isProcessing: false,
+            allLockEvents: [],
+            error: null,
+            fetchingLockData: true,
+        };
+    }
+
+    // used for fetching data periodically
+    timerInterval: any;
+
+    // get and set the web3 state when the component is mounted
+    componentDidMount = async () => {
+        const web3State = await connectWeb3(LockSeason.First);
+        this.setState(web3State);
+
+        // checks if account has changed in MetaMask
+        if ((window as any).ethereum.on) {
+            (window as any).ethereum.on('accountsChanged', this.handleAccountChange);
+        }
+
+        this.setState({ networkType: await this.state.web3.eth.net.getNetworkType() });
+
+        this.timerInterval = setInterval(() => {
+            this.getLockData().then(() => {
+                this.setState({ isLoading: false });
+            });
+        }, 1000);
+    };
+
+    isMainnet = () => {
+        return this.state.networkType === 'main';
+    };
+
+    componentWillUnmount = () => {
+        clearInterval(this.timerInterval);
+        removeWeb3Event();
+    };
+
+    // called when the user changes MetaMask account
+    handleAccountChange = () => {
+        // refresh the page
+        window.location.reload(false);
+    };
+
+    getLockData = async () => {
+        try {
+            // get all the lock events from the chain
+            const allLocks = await getAllLockEvents(this.state.web3, this.state.contract);
+
+            this.setState({ allLockEvents: allLocks });
+        } catch (error) {
+            this.setState({ error });
+        }
+    };
+
+    handleSubmit = async (formInputVal: LockInput) => {
+        // checks user input
+        if (formInputVal.amount > new BN(0) && formInputVal.duration) {
+            this.setState({ isProcessing: true });
+            //console.log(formInputVal);
+            // return a default address if user input is empty
+            const introducer = defaultAffiliation(formInputVal.affiliation).toLowerCase();
+            const { accounts, contract } = this.state;
+            try {
+                // check user input
```

Comment:
This checking algorithm is used in several places, so it is better to bundle it as a function that returns error messages.

Reply:
fixed https://github.com/staketechnologies/lockdrop-ui/pull/53/commits/320aa22aee26776a770c61060a1085a128d6979c